nifi-nar-bundles/nifi-jms-bundle/nifi-jms-processors/src/test/java/org/apache/nifi/jms/processors/helpers/AssertionUtils.java

```diff
@@ -0,0 +1,66 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.nifi.jms.processors.helpers;
+
+import org.apache.commons.lang3.exception.ExceptionUtils;
+
+import java.util.List;
+
+import static org.junit.jupiter.api.Assertions.fail;
+
+public class AssertionUtils {
+
+    public static <T extends Throwable> void assertCausedBy(Class<T> expectedType, Runnable runnable) {
+        assertCausedBy(expectedType, null, runnable);
+    }
+
+    public static <T extends Throwable> void assertCausedBy(Class<T> expectedType, String expectedMessage, Runnable runnable) {
+        try {
+            runnable.run();
+            fail(String.format("Expected an exception to be thrown with a cause of %s, but nothing was thrown.", expectedType.getCanonicalName()));
+        } catch (Throwable throwable) {
+            final List<Throwable> causes = ExceptionUtils.getThrowableList(throwable);
+            for (Throwable cause : causes) {
+                if (expectedType.isInstance(cause)) {
+                    if (expectedMessage != null) {
+                        if (cause.getMessage() != null && cause.getMessage().startsWith(expectedMessage)) {
+                            return;
+                        }
+                    } else {
+                        return;
+                    }
+                }
+            }
+            fail(String.format("Exception is thrown but not found %s as a cause. Received exception is: %s", expectedType.getCanonicalName(), throwable), throwable);
+        }
+    }
+
+    public static void assertCausedBy(Throwable expectedException, Runnable runnable) {
```

Comment:
I provided this overload because checking the exact exception is much safer. When I mock the exception, why should I check if it is "similar" to what I expect when I can check if it is precisely the mocked exception?